### ISSUE TITLE
Template format error: Parameter count 61 is greater than max allowed 60.

### ISSUE THREAD

**Issue by mharini** [dougmoscrop/serverless-plugin-split-stacks#123]
Hi,

My split stack configuration is as below:

splitStacks:
    perGroupFunction: true
    nestedStackCount: 20

I also have  versionFunctions: false

I am getting the error:
`Template format error: Parameter count 61 is greater than max allowed 60.`

What configuration should I change to fix this issues? Please help!

**Comment by dougmoscrop** [comment]
Sorry at this point you have to audit your cloudformation-update template to see which one has the most params.. you could write a very very small .js script to do this for you (also I'd love a PR that outputs this information!)

**Comment by mharini** [comment]
In serverless.yml, There are 61 functions defined. That is the source of this.

Can you please elaborate more on the .js script you've mentioned?

**Comment by dougmoscrop** [comment]
just something like:

const template = require('./serverless/cloudformation-update.json')
Object.entries(template.Resources).forEach(resource => // some debug logic

**Comment by mharini** [comment]
Okay.. So now, the functions defined in serverless.yml is the reason behind this. Is there any solution or workaround to solve this issue?

**Comment by morficus** [comment]
this issue seems to be the same as #91

**Comment by morficus** [comment]
@mharini are you use AppSync by any chance?

**Comment by dougmoscrop** [comment]
Patching to support parameter count as a hint to start a new stack should not be difficult. I might have time tonight to look at it.

**Comment by mharini** [comment]
@morficus No. I am not using AppSync.

**Comment by vicary** [comment]
So while we wait for @dougmoscrop, one of my teammates came up with a somewhat working snippet.

```js
// lib/utils.js#stackHasRoom

  stachHasRoom(stack) {
    const resources = stack.Resources || {};
    const outputs = stack.Outputs || {};
    const referenceCount = Object.values(resources).reduce(
      (prev, resource) => prev + this.getReferencedResources(resource).length, 
      0
    );

    return Object.keys(resources) < 200
      && Object.keys(outputs) < 60
      && referenceCount < 60;
  }
```

Not sure if resource references are the only things get parameterized though. If this sounds like a right direction, maybe we can start a PR?

**Comment by morficus** [comment]
@dougmoscrop ping - any chance we could get this patched in a v1.10?

**Comment by morficus** [comment]
Trying to ping @dougmoscrop once more, to see if the above solution is acceptable and if he would be interested in a PR

**Comment by dougmoscrop** [comment]
If you can PR this and it has tests and works I will happily release it!

**Comment by vicary** [comment]
@dougmoscrop hey thanks for the reply. I can make a PR but really needs your input since I'm not super familiar with the way CloudFormation works.

My snippet above will resulting in some of the nested stacks containing only 20 resources because of the high parameter count, if this sounds like an authentic limitation of the SAM model syntax I'll start a PR.

**Comment by dougmoscrop** [comment]
Yeah, exactly - not even SAM, it's just plain CloudFormation limits.

**Comment by morficus** [comment]
Fun fact: AWS has raised the template parameter limit from 60 to 200!

https://aws.amazon.com/about-aws/whats-new/2020/10/aws-cloudformation-now-supports-increased-limits-on-five-service-quotas/
